### PR TITLE
[plugin-azure-cosmos-db] Introduce plugin to work with Azure Cosmos DB

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -153,7 +154,8 @@ jobs:
                                                 -Pvividus.allure.executor.build-name="System Tests ${GITHUB_RUN_ID}" \
                                                 -Pvividus.allure.executor.build-url=https://github.com/vividus-framework/vividus/actions/runs/${GITHUB_RUN_ID} \
                                                 -Pvividus.allure.executor.report-url=https://github.com/vividus-framework/vividus/actions/runs/${GITHUB_RUN_ID} \
-                                                -Pvividus.allure.executor.report-name="System tests report"
+                                                -Pvividus.allure.executor.report-name="System tests report" \
+                                                -pvividus.cosmos-db.test.key=${AZURE_COSMOS_DB_KEY}
```
Comment (valfirst) [inline .github/workflows/gradle.yml:158]: `-P...`

File: build.gradle [hunk 2]
```diff
@@ -73,6 +73,7 @@ ext {
         commonsLang3:           '3.12.0',
         commonsText:            '1.9',
         commonsCollections4:    '4.4',
+        equalsverifier:         '3.5.5',
```
Comment (valfirst) [inline build.gradle:76]: it's better to keep the version copy-pasted in different modules, dependabot can update all of them within one PR (see, AWS modules as examples)

File: vividus-plugin-azure-cosmos-db/src/main/resources/properties/defaults/default.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+azure.storage.queue.receive-timeout=PT1M
```
Comment (valfirst) [inline vividus-plugin-azure-cosmos-db/src/main/resources/properties/defaults/default.properties:1]: to be removed?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-azure-cosmos-db.adoc [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+}` into CosmosDB container `test`
+----
+
+=== Delete
+
+Delete an item in the container
+
+[source,gherkin]
+----
+When I insert `$data` into CosmosDB container `$containerKey`
```
Comment (abudevich) [inline docs/modules/plugins/pages/plugin-azure-cosmos-db.adoc:164]: Are you sure that this wording of step for deleting ?

File: vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbSteps.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            CosmosDbService cosmosDbService, SoftAssert softAssert)
+    {
+        this.bddVariableContext = bddVariableContext;
+        this.containers = containers;
+        this.cosmosDbService = cosmosDbService;
+        this.softAssert = softAssert;
+    }
+
+    /**
+     * Reads an item by its id and partition key and saves result into scoped variable
```
Comment (abudevich) [inline vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbSteps.java:50]: the result?

File: docs/modules/plugins/pages/plugin-azure-cosmos-db.adoc [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+----
+
+Where:
+
+. `container-name` - The name of the container
+. `property-name` - The name of the container property. One of: 
+.. `key` - The key to access database
+.. `endpoint` - The database endpoint
+.. `database-id` - The id of the database
+.. `container-id` - The id of the container
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-cosmos-db.adoc:31]: the docs should be updated to reflect the non-flat approach in the properties

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbService.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        this.databases = databases;
+    }
+
+    public String executeQuery(CosmosDbContainer cosmosDbContainer, String query)
+    {
+        List<JsonNode> result = containers.getUnchecked(cosmosDbContainer)
+                                                  .queryItems(query, null, JsonNode.class)
+                                                  .stream()
+                                                  .collect(Collectors.toList());
+        return jsonUtils.toPrettyJson(result);
```
Comment (uarlouski) [inline vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbService.java:78]: Collectors collectingAndThen ? :)

File: vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbSteps.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+import org.vividus.bdd.context.BddVariableContext;
+import org.vividus.bdd.variable.VariableScope;
+import org.vividus.softassert.SoftAssert;
+import org.vividus.util.property.PropertyMappedCollection;
+
+public class CosmosDbSteps
+{
+    private static final int CREATED = 201;
+    private static final int NO_CONTENT = 204;
+    private static final int OK = 200;
```
Comment (uarlouski) [inline vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbSteps.java:33]: is it possible to use existing constants from transitive HTTP libs?

Comment (ikalinin1) [reply vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbSteps.java:33]: was not able to find anything useful, won't add extra dependency just to refer status codes

File: vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbService.java (hunk 7 above)
Comment (ikalinin1) [reply vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbService.java]: good catch!

**Review by valfirst — APPROVED** [review]

File: vividus-plugin-azure-cosmos-db/build.gradle [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    implementation project(':vividus-util')
+    implementation project(':vividus-soft-assert')
+    implementation(group: 'com.azure', name: 'azure-cosmos', version: '4.13.1')
+    implementation(group: 'com.google.guava', name: 'guava', version: versions.guava)
+
+    testImplementation platform(group: 'org.junit', name: 'junit-bom', version: versions.junit)
+    testImplementation(group: 'org.junit.jupiter', name: 'junit-jupiter')
+    testImplementation(group: 'org.mockito', name: 'mockito-junit-jupiter', version: versions.mockito)
+    testImplementation(group: 'nl.jqno.equalsverifier', name: 'equalsverifier', version: '3.5.5')
+
```
Comment (uarlouski) [inline vividus-plugin-azure-cosmos-db/build.gradle:14]: :)

**Review by uarlouski — APPROVED** [review]

**Pull request merged**